lib/logstash/outputs/elasticsearch/common_configs.rb

```diff
@@ -75,13 +75,8 @@ def self.included(mod)
       # to prevent LS from sending bulk requests to the master nodes.  So this parameter should only reference either data or client nodes in Elasticsearch.
       mod.config :hosts, :validate => :array, :default => ["127.0.0.1"]
 
-      # This plugin uses the bulk index API for improved indexing performance.
-      # This setting defines the maximum sized bulk request Logstash will make
-      # You you may want to increase this to be in line with your pipeline's batch size.
-      # If you specify a number larger than the batch size of your pipeline it will have no effect,
-      # save for the case where a filter increases the size of an inflight batch by outputting
-      # events.
-      mod.config :flush_size, :validate => :number, :default => 500
+      # DEPRECATED. We now restrict bulk sizes to a size that balances heap size with reducing round-trips. Very large events will be sent in a single request by themselves.
+      mod.config :flush_size, :validate => :number, :default => 500, :deprecated => "No Longer Needed"
```

Comment:
I agree with everything but this being a good use of deprecate.
Tagging an option as deprecated and making it a noop can be very dangerous (remember this setting is non default very frequently). I don't feel comfortable in assuming flush_size is useless for everyone and simply swapping the flush heuristic from count to size will be a win for everyone.
Is for this lack of certainty that we have the deprecate -> obsolete cycle, so we can test the waters, provide a fall back for unexpected scenarios and also a smoother migration.